synapseclient/client.py

```diff
@@ -1988,78 +1996,62 @@ def _deleteFileHandle(self, fileHandle):
     ############################################################
 
     def _getDefaultUploadDestination(self, entity):
-        return self.restGET('/entity/%s/uploadDestination'% id_of(entity),
+        return self.restGET('/entity/%s/uploadDestination'% entity['parentId'],
                      endpoint=self.fileHandleEndpoint)
 
```

Comment:
Why the change to parentId?  Was this a bug?  I.e.  did this function not actually work?